### PR TITLE
feat: added design tokens eslint rules to mobile

### LINKED ISSUE #9378: Implement `color-no-hex` Rule to Enforce Design Token Usage
### What is this about?

To ensure our mobile's codebase remains consistent with our design system and to promote better theming, accessibility, and visual consistency, we propose the implementation of the [color-no-hex](https://github.com/MetaMask/eslint-plugin-design-tokens/blob/main/docs/rules/color-no-hex.md) rule from the [@metamask/eslint-plugin-design-tokens package](https://github.com/MetaMask/eslint-plugin-design-tokens). This rule aims to prevent the use of static hex color values in our styles, encouraging instead the adoption of design tokens. By enforcing this rule, we aim to maintain a high standard of code quality and design system alignment, facilitating easier updates and ensuring a cohesive user experience.

### Scenario

_No response_

### Design

_No response_

### Technical Details

- **Objective**: Integrate the `color-no-hex` rule into the mobile's ESLint configuration to discourage the use of static hex color values and promote the use of design tokens.
- **Implementation Steps**:
  1. Ensure the `eslint-plugin-design-tokens` package is installed and up-to-date in the mobile's project.
  2. Update the `.eslintrc` configuration file to include the `color-no-hex` rule, setting it to "warn"
  3. Run ESLint across the mobile's codebase to identify instances where static hex color values are currently used.

### Threat Modeling Framework

_No response_

### Acceptance Criteria

- The `color-no-hex` rule is successfully integrated into the mobile's ESLint setup and actively flags the use of static hex color values.
- All valid instances of static hex color values in the codebase are ignored e.g. image asset svgs components
- The mobile's documentation and developer guidelines are updated to reflect the importance of using design tokens for color definitions.

### Stakeholder review needed before the work gets merged

- [X] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by brianacnguyen**
## **Description**
This PR adds the color-no-hex lint rules from the design-tokens repo. The purpose of this rule is to discourage developers from using hardcoded hex colors, promoting the use of color tokens defined by the design system team. This change ensures consistency in color usage across the application and aligns with the design system guidelines.

## **Related issues**

Fixes: #9378 

## **Manual testing steps**

1. Try to use hex colors in any file
2. Save
3. Observe lint in action

### **After**
https://github.com/MetaMask/metamask-mobile/assets/8112138/bdcf19be-3e0f-47b2-a993-dbcfd80a8cc3
- Lint error when a hex color is used
Screenshot 2024-06-04 at 10 24 37 AM

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -39,6 +39,12 @@ module.exports = {
         'import/no-nodejs-modules': 0,
       },
     },
+    {
+      files: ['**/*.test.{js,ts,tsx}', '**/*.stories.{js,ts,tsx}'],
+      rules: {
+        '@metamask/design-tokens/color-no-hex': 'off',
+      },
+    },
```
Comment (georgewrmarshall) [inline .eslintrc.js:47]: Ignores tests and stories

File: .eslintrc.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ module.exports = {
     'plugin:import/warnings',
     'plugin:react/recommended',
   ],
-
+  plugins: ['@metamask/design-tokens'],
```
Comment (georgewrmarshall) [inline .eslintrc.js:14]: This is what you were missing @brianacnguyen 

Screenshot 2024-05-16 at 4 47 02 PM

File: app/components/UI/Ramp/components/LoadingAnimation/backgroundShapes.ts [hunk 3]
```diff
@@ -1,3 +1,5 @@
+// Intentionally using non-design system colors here
+/* eslint-disable @metamask/design-tokens/color-no-hex*/
```
Comment (georgewrmarshall) [inline app/components/UI/Ramp/components/LoadingAnimation/backgroundShapes.ts:2]: Ignoring as it's an image asset component

File: app/components/UI/Swaps/components/LoadingAnimation/backgroundShapes.js [hunk 4]
```diff
@@ -1,3 +1,5 @@
+// Intentionally using non-design system colors here
+/* eslint-disable @metamask/design-tokens/color-no-hex*/
```
Comment (georgewrmarshall) [inline app/components/UI/Swaps/components/LoadingAnimation/backgroundShapes.js:2]: Ignoring as it's an image asset component

File: app/util/theme/temp-tokens/brandColors.ts [hunk 5]
```diff
@@ -1,6 +1,7 @@
 /*eslint no-useless-computed-key: "error"*/
 /* eslint-disable no-useless-computed-key */
 /* eslint-disable import/prefer-default-export */
+/* eslint-disable @metamask/design-tokens/color-no-hex*/
 
 import { BrandColors } from './brandColors.types';
```
Comment (georgewrmarshall) [inline app/util/theme/temp-tokens/brandColors.ts:6]: The `brandColors` object in this file can now point to the `brandColors` object from the design tokens in the v4 upgrade

**Review by georgewrmarshall — DISMISSED** [review]
LGTM 🚀 🎨

File: app/util/networks/index.js [hunk 6]
```diff
@@ -97,12 +105,16 @@ export const NetworkList = {
     networkId: 59141,
     chainId: toHex('59141'),
     color: '#61dfff',
+    // Third party color
+    // eslint-disable-next-line @metamask/design-tokens/color-no-hex
```
Comment (georgewrmarshall) [inline app/util/networks/index.js:109]:
```suggestion
    // Third party color
    // eslint-disable-next-line @metamask/design-tokens/color-no-hex
    color: '#61dfff',
```

**Review by georgewrmarshall — COMMENTED** [review]
One small change after merge from main!

File: app/util/theme/temp-tokens/brandColors.ts (hunk 5 above)
Comment (brianacnguyen) [reply app/util/theme/temp-tokens/brandColors.ts:6]: yup!

**Review by georgewrmarshall — APPROVED** [review]

**Review by NicolasMassart — APPROVED** [review]
Looks good to me.

**Pull request merged**